server/api/src/dtos/expenses.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{transaction, Id, MoneyAmount};
+use entity::expense;
+use serde::{self, Deserialize, Serialize};
+
+pub type ExpenseResponse = Vec<ExpenseWithTransactions>;
+
+struct ExpenseWithTransactions {
+    pub id: Id,
+    pub name: String,
+    pub description: String,
+    pub value: MoneyAmount,
+    pub start_date: Date,
+    pub user_id: Id,
+    pub currency_type_id: Id,
+    pub recurrence_type_id: Id,
+    pub predefined_expense_id: Option<Id>,
+    pub transactions: Vec<transaction::Model>,
+}
+
+impl ExpenseWithTransactions {
+    fn from(&self, expense: expense::Model, transactions: Vec<transaction::Model>) -> self {
```

Comment:
This from function somehow feels odd, I will search for a way to make it more idiomatic, good enough for now tough.

Reply:
This leftover file is getting deleted.